### PR TITLE
Add Game Type value rule for format override

### PR DESCRIPTION
**Pull request by TurboRx**
https://www.smogon.com/forums/threads/forcedoubles-clause-for-buildformat-and-challenge-codes.3764216/

### CODE REVIEW

File: data/rulesets.ts [hunk 1]
```diff
@@ -2070,6 +2070,21 @@ export const Rulesets: import('../sim/dex-formats').FormatDataTable = {
...
 		},
 	},
+	gametype: {
+		effectType: 'Rule',
+		name: 'Game Type',
+		desc: "Overrides the format's default game type (Singles, Doubles, Triples, Multi, or Free-For-All)",
+		hasValue: true,
+		onValidateRule(value) {
+			const gameType = this.dex.toID(value);
+			const validGameTypes = ['singles', 'doubles', 'triples', 'rotation', 'multi', 'freeforall'];
```
Comment (Slayer95) [inline data/rulesets.ts]: Rotation is not (yet) supported.

Comment (TurboRx) [reply data/rulesets.ts]: ok, now removed it.

**Comment by TurboRx** [comment]
that's a valid concern. let me add some validation to check for incompatible combinations. i'll test a few edge cases and push an update.

File: data/rulesets.ts [hunk 2]
```diff
@@ -2081,6 +2081,19 @@ export const Rulesets: import('../sim/dex-formats').FormatDataTable = {
 			if (!validGameTypes.includes(gameType)) {
 				throw new Error(`Invalid game type "${value}". Valid game types are: ${validGameTypes.join(', ')}`);
 			}
+
+			// validate compatibility with other rules
+			if (this.ruleTable.has('chimera1v1rule')) {
```
[the remark this answers is not in the document]
Comment (Clastia) [reply data/rulesets.ts]: shared power is the only other exception that i know of (specifically it should just not allow challenging when gameType!=='singles'), but im working on that issue myself

[the remark this answers is not in the document]
Comment (Slayer95) [reply data/rulesets.ts]: I think ``Chimera 1v1 Rule`` should (already) be handling game type validation in its own ``onValidateRule`` handler. Similarly, formats with custom built-in behavior not coming from a ``Rule`` should be made incompatible with ``Game Type = X`` going forwards.

File: data/rulesets.ts [hunk 3]
```diff
@@ -2070,6 +2070,34 @@ export const Rulesets: import('../sim/dex-formats').FormatDataTable = {
...
+				if (gameType === 'doubles' || gameType === 'triples') {
+					throw new Error(`Game Type = ${value} is incompatible with Chimera 1v1 (sides can have multiple active Pokémon).`);
+				}
+			}
+			if (this.ruleTable.has('bestof')) {
+				if (!['singles', 'doubles'].includes(gameType)) {
+					throw new Error(`Game Type = ${value} is incompatible with Best Of (only singles and doubles are supported).`);
```
Comment (Slayer95) [inline data/rulesets.ts]: ?? Why?

Comment (Slayer95) [reply data/rulesets.ts]: Why is Best-Of incompatible with Triples?

**Comment by TurboRx** [comment]
done. fixed this problems, tested locally now it's working

**Comment by TurboRx** [comment]
IMG_20251222_130018
IMG_20251222_130040
IMG_20251222_130119
IMG_20251222_130136
IMG_20251222_130217
IMG_20251222_130239
IMG_20251222_130959

**Comment by TurboRx** [comment]
triples is included because it's one of the five supported game types (singles, doubles, triples, multi, freeforall). that's why I thought including it is better. if you want me to remove it then I'll remove it.

File: test/sim/misc/gametype-rule.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+		assert.throws(
+			() => common.createBattle({ customRules: ['Chimera 1v1 Rule', 'Game Type = Triples'] }),
+			/incompatible with Chimera 1v1|cannot be 1v1/
+		);
+	});
+
+	it('should reject non-singles/doubles with Best Of', () => {
+		battle = null;
+		assert.throws(
+			() => common.createBattle({ customRules: ['Game Type = Triples', 'Best Of = 3'] }),
```
Comment (Slayer95) [inline test/sim/misc/gametype-rule.js]: Please remove this assertion per @DaWoblefet comment in #9793.

File: test/sim/misc/gametype-rule.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	it('should reject non-singles game types with format-specific restrictions', () => {
+		battle = null;
+		// Shared Power format has onValidateRule that only allows singles
+		const Dex = require('../../../dist/sim/dex').Dex;
+		const BattleClass = require('../../../dist/sim/battle').Battle;
+
+		assert.throws(
+			() => {
+				const format = Dex.formats.get('[Gen 9] Shared Power @@@Game Type = Doubles', true);
+				new BattleClass({ format });
```
[the remark this answers is not in the document]
Comment (TurboRx) [reply test/sim/misc/gametype-rule.js]: oh, thanks, i missed that by mistake.

[the remark this answers is not in the document]
Comment (TurboRx) [reply test/sim/misc/gametype-rule.js]: now fixed.

**Comment by HisuianZoroark** [comment]
Multi Battles and Free For Alls started via this rule crash the server when I tested, but thankfully no OMs have been broken with this rule

**Comment by TurboRx** [comment]
ok, now fixed it.
